src/metabase/db/liquibase.clj

```diff
@@ -496,27 +516,13 @@
     ;; count and rollback only the applied change set ids which come after the target version (only the "v..." IDs need to be considered)
      (let [changeset-query (format "SELECT id FROM %s WHERE id LIKE 'v%%' ORDER BY ORDEREXECUTED ASC" (changelog-table-name liquibase))
            changeset-ids   (map :id (jdbc/query {:connection conn} [changeset-query]))
-          ;; IDs in changesets do not include the leading 0/1 digit, so the major version is the first number
-           ids-to-drop     (drop-while #(not= (inc target-version) (first (extract-numbers %))) changeset-ids)]
+           ;; IDs in changesets do not include the leading 0/1 digit, so the major version is the first number
+           ids-to-drop     (drop-while #(not= (inc target-version) (first (extract-numbers %))) changeset-ids)
+           latest-available (latest-available-major-version liquibase)
+           latest-applied   (latest-applied-major-version conn (.getDatabase liquibase))]
+       (when (> latest-applied latest-available)
+         (throw (IllegalArgumentException.
```

Comment:
Not sure I'd call this an "Illegal Argument" - maybe just a generic `ex-info` would make sense?

Reply:
Yeah, same general logic just in a different spot. Keeping someone from downgrading when they're on a too-old version vs. starting up when they're on a too-old version